Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverridePublicInterface.h

```diff
@@ -34,12 +34,23 @@ namespace AzToolsFramework
             //! by the class implmenting this interface based on certain selections in the editor. eg: the prefab currently being edited.
             //! @param entityId The id of the entity for which to get the override type.
             //! @return an override type if an override exists on the given entity id.
-            virtual AZStd::optional<OverrideType> GetOverrideType(AZ::EntityId entityId) = 0;
-            
+            virtual AZStd::optional<OverrideType> GetEntityOverrideType(AZ::EntityId entityId) = 0;
+
+            //! Gets the override type on the given component. The prefab that creates the overrides is identified
+            //! by the class implmenting this interface based on certain selections in the editor. eg: the prefab currently being edited.
+            //! @param component The component for which to get the override type.
+            //! @return an override type if an override exists on the given component.
+            virtual AZStd::optional<OverrideType> GetComponentOverrideType(const AZ::Component& component) = 0;
+
             //! Revert overrides on the entity matching the entity id. Returns false if no overrides are present on the entity.
             //! @param entityId The id of the entity on which overrides should be reverted.
             //! @return Whether overrides are successfully reverted on the entity.
             virtual bool RevertOverrides(AZ::EntityId entityId) = 0;
+
+            //! Revert overrides on the given component. Returns false if no overrides are present on the component.
+            //! @param component The component on which overrides should be reverted.
+            //! @return Whether overrides are successfully reverted on the component.
+            virtual bool RevertComponentOverrides(const AZ::Component& component) = 0;
```

Comment:
The idea is to keep this interface scriptable so that we can write python automated tests by calling this. As such, the parameters of the functions here should be usable in tests. I don't think AZ::Component is a type that python tests can interpret. Since we only care about GetSerializedIdentifier() within the component, I suggest you pass that as a string to these new functions.

Another alternative could be to pass the component id and fetch the component from the entity using the id. However, you'd have to jump multiple hoops to get to the serialized identifier in that case. So the above approach would be better imo

Reply:
I thought about using component Id and using Entity::FindComponent() to find the component. The "find" method isn't that efficient though, but it might be OK enough for this use case. I'll look into using the serialized identifier as well.

Reply:
Changed to use EntityComponentIdPair.